### PR TITLE
pkg/daemon: idiomatic Go cleanup

### PR DESCRIPTION
**Pull request by runcom**
Went through a bunch of cleanup to be more idiomatic when writing Go
code (no else if previous return, default values, return on err, bool comparison etc etc)
(mainly went to read and learn the code)
There should be no functional change at all.

(so sorry for this PR :angel: )

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
@runcom this is the best thank you!  checking it out!

File: pkg/controller/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	kscheme "k8s.io/client-go/kubernetes/scheme"
 
-	"github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	v1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
```
Comment (abhinavdahiya) [inline pkg/controller/bootstrap/bootstrap.go]: Why this change?

File: pkg/daemon/update.go [hunk 2]
```diff
@@ -303,16 +299,10 @@ func (dn *Daemon) updateFiles(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	if err := dn.writeFiles(newConfig.Spec.Config.Storage.Files); err != nil {
 		return err
 	}
-
 	if err := dn.writeUnits(newConfig.Spec.Config.Systemd.Units); err != nil {
 		return err
 	}
-
-	if err := dn.deleteStaleData(oldConfig, newConfig); err != nil {
```
Comment (abhinavdahiya) [inline pkg/daemon/update.go]: nit: if in future you add another step after deleteStaleData, you'll have to bring back this if... The current setup keeps it clearer imo.

File: pkg/daemon/writer.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import (
 	"fmt"
 
 	"github.com/golang/glog"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (abhinavdahiya) [inline pkg/daemon/writer.go]: Why this? Maybe `corev1`?

Comment (runcom) [reply pkg/daemon/writer.go]: This is golint doing this, overlooked

File: pkg/daemon/update.go (hunk 2 above)
Comment (runcom) [reply pkg/daemon/update.go]: Boils down to preference for me, if you prefer it that way I'll change it back

File: pkg/controller/bootstrap/bootstrap.go (hunk 1 above)
Comment (runcom) [reply pkg/controller/bootstrap/bootstrap.go]: Golint doing stupid stuff

**Comment by jlebon** [comment]
Very nice!
/lgtm

File: pkg/daemon/writer.go [hunk 4]
```diff
@@ -103,9 +103,9 @@ func (nw *NodeWriter) SetUpdateDegraded(err error, client corev1.NodeInterface,
 // node to degraded due to that error.
 func (nw *NodeWriter) SetUpdateDegradedIgnoreErr(err error, client corev1.NodeInterface, node string) error {
 	// log error here since the caller won't look at it
-	degraded_err := nw.SetUpdateDegraded(err, client, node)
-	if degraded_err != nil {
-		glog.Errorf("Error while setting degraded: %v", degraded_err)
+	degradedErr := nw.SetUpdateDegraded(err, client, node)
```
Comment (yuqi-zhang) [inline pkg/daemon/writer.go]: To keep it consistent we should probably just make it `if degradedErr := nw.SetUpdateDegraded(err, client, node); degradedErr != nil` here right?
